pyomo/gdp/plugins/bigm.py

```diff
@@ -1004,3 +1004,33 @@ def get_M_value(self, constraint):
         # fails... (That is, it's a bug in the mapping.)
         lower, upper = transBlock.bigm_src[constraint]
         return (lower[0], upper[0])
+
+    def get_all_M_values_by_constraint(self, model):
+        """Returns a dictionary mapping each constraint to a tuple:
+        (lower_M_value, upper_M_value), where either can be None if the
+        constraint does not have a lower or upper bound (respectively).
+
+        Parameters
+        ----------
+        model: A GDP model that has been transformed with BigM
+        """
+        m_values = {}
+        for disj in model.component_data_objects(
+                Disjunct,
+                active=None,
+                descend_into=(Block, Disjunct)):
+            if disj.transformation_block is not None:
+                transBlock = disj.transformation_block()
+                for cons, (lower, upper) in transBlock.bigm_src.items():
+                    m_values[cons] = self.get_M_value(cons)
+        return m_values
+
+    def get_largest_M_value(self, model):
+        """Returns the largest M value for any constraint on the model.
```

Comment:
Why does this return a scalar and not a 2-tuple of `(biggest_lower_M, biggest_upper_M)`?

Reply:
Do you think that is really safe? It's useful if you're not going to change anything on the model, but if for some reason you rewrote some things to make an equivalent model, what's lower and what's upper could change. I just worry because that's not really a mathematical distinction so much as a how-the-constraint-expressions-are-represented distinction.

Reply:
Yeah - good point.  Plus, if the user wants to calculate the tuple, it is not difficult to do that using `get_all_M_values_by_constraint()`